recipes-core/initrdscripts/initramfs-stubdomain/init.sh

```diff
@@ -97,6 +97,8 @@ device_model="device-model/$target"
 echo "Starting vchan-socket-proxy"
 vchan-socket-proxy -m server 0 $device_model/qmp-vchan /tmp/qemu.qmp &
 
-wait
-
-poweroff -f
+while true; do
```

Comment:
+1 for a stubdom shell. I've done this successfully in dev environments by adding:

`/sbin/getty 115200 hvc0 -n -l /bin/sh &`

right before qemu is called. This did not require cttyhack or setsid and can be utilized with a simple `xl console <dm_name>`. Is there a reason behind the ctty hack? I'm not too familiar with it.

Reply:
cttyhack is take from Qubes.  The qubes way of doing things made it upstream to Xen, so aligning the approach seems preferable.  Also, this way will respawn a new shell if the first is closed.

From https://github.com/brgl/busybox/blob/master/shell/cttyhack.c
cttyhack avoids the "can't access tty; job control turned off" error message, which typically appears when one tries to use a shell with stdin/stdout on /dev/console.

Reply:
Okay, if it made it upstream then it's good enough for me